crates/uv/tests/it/sync.rs

```diff
@@ -5527,6 +5527,96 @@ fn sync_git_repeated_member_backwards_path() -> Result<()> {
     Ok(())
 }
 
+/// A Git repository that points to a pre-built archive within the repository.
+#[test]
+fn sync_git_path_archive() -> Result<()> {
```

Comment:
This is the real motivating example: you have a project that you push to GitHub that includes a `path` source that points to a wheel, like:

```toml
[project]
name = "archive-in-git-test"
version = "0.1.0"
description = "Add your description here"
readme = "README.md"
requires-python = ">=3.13"
dependencies = ["iniconfig"]

[tool.uv.sources]
iniconfig = { path = "archives/iniconfig-2.0.0-py3-none-any.whl" }
```

We have to translate that path source to a Git source so that we avoid writing cache paths to the lockfile.